### ISSUE TITLE
Don't get any device

### ISSUE THREAD

**Issue by grodansparadis** [rusoku/TouCAN-SocketCAN#11]
On the latest Ubuntu (kernel 5.15.0-37-generic) when I do a new install of the toucan driver it complains about SSL 
here

arch/x86/Makefile:142: CONFIG_X86_X32 enabled but no binutils support
  INSTALL /lib/modules/5.15.0-37-generic/extra/toucan.ko
  SIGN    /lib/modules/5.15.0-37-generic/extra/toucan.ko
At main.c:160:
- SSL error:FFFFFFFF80000002:system library::No such file or directory: ../crypto/bio/bss_file.c:67
- SSL error:10000080:BIO routines::no such file: ../crypto/bio/bss_file.c:75
sign-file: certs/signing_key.pem: No such file or directory
  DEPMOD  /lib/modules/5.15.0-37-generic
Warning: modules_install: missing 'System.map' file. Skipping depmod.
make[1]: Leaving directory '/usr/src/linux-headers-5.15.0-37-generic'

but still compiles fine and I can load the driver.

$ sudo lsmod | grep toucan

gives

toucan                 20480  0
can_dev                36864  2 usb_8dev,toucan

but I don't get any can device for toucan (expected can1 to be available after insmod).

Have the same problem also if the  usb2can is not present.

Need some hints on how to solve this.

**Comment by rusoku** [comment]
Hi Ake, it works for me:

```
ged@VirtualBoxPC:~/Dev/TouCAN-SocketCAN$ uname -r
5.15.0-40-generic
ged@VirtualBoxPC:~/Dev/TouCAN-SocketCAN$ lsmod | grep toucan
toucan                 20480  0
can_dev                36864  1 toucan
ged@VirtualBoxPC:~/Dev/TouCAN-SocketCAN$ ifconfig -a
can0: flags=128<NOARP>  mtu 16
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 10  (UNSPEC)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

```

**Comment by rusoku** [comment]
make toucan
sudo cp ./toucan.ko /lib/modules/$(uname -r)/kernel/drivers/misc
sudo depmod -a
sudo modprobe toucan

**Comment by grodansparadis** [comment]
I get the same result. 

$ ls /sys/class/net/

gives

can0  enp6s0  lo

where can0 is the 8dev  canusb adapter.

**Comment by grodansparadis** [comment]
akhe@fluorine:~/development/TouCAN-SocketCAN$ modinfo toucan

filename:       /lib/modules/5.15.0-37-generic/extra/toucan.ko
license:        GPL v2
description:    CAN driver for Rusoku Technologies TouCAN adapters
srcversion:     068B9A78C02003D26E61129
alias:          usb:v16D0p0EACd*dc*dsc*dp*ic*isc*ip*in*
depends:        can-dev
retpoline:      Y
name:           toucan
vermagic:       5.15.0-37-generic SMP mod_unload modversions

**Comment by rusoku** [comment]
With attached 2x TouCAN adapters and Peak FD pro dual adapter I get four CAN interfaces:

`ged@VirtualBoxPC:~$ ifconfig -a
can0: flags=128<NOARP>  mtu 16
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 10  (UNSPEC)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

can1: flags=128<NOARP>  mtu 16
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 10  (UNSPEC)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

can2: flags=128<NOARP>  mtu 16
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 10  (UNSPEC)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

can3: flags=128<NOARP>  mtu 16
        unspec 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  txqueuelen 10  (UNSPEC)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
`

**Comment by rusoku** [comment]
ged@VirtualBoxPC:~$ ls /sys/class/net
can0  can1  can2  can3  enp0s3  lo

**Comment by rusoku** [comment]
ged@VirtualBoxPC:~$ modinfo toucan
filename:       /lib/modules/5.15.0-40-generic/kernel/drivers/misc/toucan.ko
license:        GPL v2
description:    CAN driver for Rusoku Technologies TouCAN adapters
srcversion:     068B9A78C02003D26E61129
alias:          usb:v16D0p0EACd*dc*dsc*dp*ic*isc*ip*in*
depends:        can-dev
retpoline:      Y
name:           toucan
vermagic:       5.15.0-40-generic SMP mod_unload modversions

**Comment by grodansparadis** [comment]
When I connect both canusb and a peak adapter I also get two interfaces.  But no toucan.

modinfo -F intree toucan

gives nothing. Do yo get 'Y' there?

**Comment by rusoku** [comment]
modinfo -F intree toucan

gives nothing as well

**Comment by rusoku** [comment]
IMHO because TouCAN is out-of-tree module

**Comment by grodansparadis** [comment]
Yes of course.  That is correct. My mistake.

It is so strange as everything else looks OK except that the name does not come up.

**Comment by grodansparadis** [comment]
Tested on one of my servers also with kernel 4.19.0-20-amd64 and also on a Pi and get the same result.  Very strange.  But at least the behavior is consistent. I will have vacation for some weeks from Friday and will check this further after that.

**Comment by rusoku** [comment]
Something wrong with adapter firmware ?

**Comment by grodansparadis** [comment]
I will try to update it.

**Comment by grodansparadis** [comment]
Is it the same steps as for the 8dev adapter? What drivers do I need to have installed on windows?

**Comment by rusoku** [comment]
No need to install any drivers for Windows.Just plug and check:  "Device Manager"-> Universal Serial Bus devices.
![TouCAN](https://user-images.githubusercontent.com/46974254/176472314-4403dd07-23be-449b-94fc-9c793c9b2964.JPG)

btw you can upgrade device on Linux or OSX machine using dfu-util
p.s. I sent out another one TouCAN device for you today.Two devices is better than one ! ;-)

**Comment by grodansparadis** [comment]
A firmware update did the trick. Strange, last time I used it it was working flawlessly but on a Raspberry Pi and then it has just been sitting on my desk for some time.  Cosmic rays probably :))

For other Linux users needing firmware updates

dfu-util --list

gives something like

Found DFU: [0483:df11] ver=0200, devnum=42, cfg=1, intf=0, path="2-1.1.4.2", alt=1, name="@Internal eeprom /0x08010000/01*64Kg", serial="3859B33F3437"
Found DFU: [0483:df11] ver=0200, devnum=42, cfg=1, intf=0, path="2-1.1.4.2", alt=0, name="@Internal Flash /0x08020000/07*128Kg", serial="3859B33F3437"

Update firmware with

dfu-util --device 0483:df11 -i 0 --alt 0 --download F4FS1_FLASH_v1.0.5.dfu  --reset 

where "F4FS1_FLASH_v1.0.5.dfu" is the firmware file.

Another adapter in the mail? You are to kind Gediminas. Will be a lot of can this autumn so it definitely will come to use.

Thanks for your kind assistance!
